src/supplemental/quic/msquic_dial.c

```diff
@@ -490,24 +498,27 @@ quic_error(void *arg, int err)
 
 	nni_mtx_lock(&c->mtx);
 	// only close aio of this stream
-	while (((aio = nni_list_first(&c->readq)) != NULL) ||
-	    ((aio = nni_list_first(&c->writeq)) != NULL)) {
+	while ((aio = nni_list_first(&c->writeq)) != NULL) {
+		nni_aio_list_remove(aio);
+		nni_aio_finish_error(aio, err);
+	}
+	while ((aio = nni_list_first(&c->readq)) != NULL) {
```

Comment:
Emmm I can't get any difference from the code before

Reply:
actually you are right